modules/sfp_threatjammer.py

```diff
@@ -0,0 +1,205 @@
+            "MALICIOUS_IPADDR",
+            "MALICIOUS_AFFILIATE_IPADDR",
+        ]
+
+    def queryIp(self, ip):
+        """Query API for an IPv4 or IPv6 address.
+
+        Args:
+            ip (str): IP address
+
+        Returns:
+            str: API response as JSON
+        """
+
+        headers = {
+            'Authorization': f"Bearer {self.opts['api_key']}",
+            'Accept': 'application/json',
+        }
+
+        res = self.sf.fetchUrl(
+            f"https://dublin.api.threatjammer.com/v1/assess/ip/{ip}",
+            timeout=self.opts['_fetchtimeout'],
+            useragent=self.opts['_useragent'],
+            headers=headers
+        )
+
+        time.sleep(1)
+
+        if res['code'] == '429':
+            self.error("You are being rate-limited by ThreatJammer.com")
+            self.errorState = True
+            return None
+
+        if res['code'] != "200":
+            self.error("Error retrieving search results from ThreatJammer.com")
+            self.errorState = True
+            return None
+
+        if res['content'] is None:
+            self.error("Received no content from ThreatJammer.com")
+            self.errorState = True
+            return None
+
+        try:
+            return json.loads(res['content'])
+        except Exception as e:
+            self.debug(f"Error processing JSON response: {e}")
+            return None
+
+        return None
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        self.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.error(
+                f"You enabled {self.__class__.__name__} but did not set an API key!"
+            )
+            self.errorState = True
+            return
+
+        if eventData in self.results:
+            self.debug(f"Skipping {eventData}, already checked.")
+            return
+
+        self.results[eventData] = True
+
+        if eventName.startswith("AFFILIATE") and not self.opts['checkaffiliates']:
+            return
+
+        if eventName in ['IP_ADDRESS', 'IPV6_ADDRESS']:
+            blacklist_type = "BLACKLISTED_IPADDR"
+            malicious_type = 'MALICIOUS_IPADDR'
+        elif eventName in ['AFFILIATE_IPADDR', 'AFFILIATE_IPV6_ADDRESS']:
+            blacklist_type = "BLACKLISTED_AFFILIATE_IPADDR"
+            malicious_type = 'MALICIOUS_AFFILIATE_IPADDR'
+        else:
+            self.debug(f"Unexpected event type {eventName}, skipping")
+            return
+
+        self.debug(f"Checking maliciousness of IP address {eventData} with ThreatJammer.com")
+
+        ip_info = self.queryIp(eventData)
+
+        if ip_info is None:
+            self.sf.error(f"Error processing JSON response for {eventData} from ThreatJammer.com")
+            return
+
+        risk_score = int(ip_info["score"])
+        risk = ip_info["risk"]
```

Comment:
These should be checked for validity.

Something like this (depending on how zero is handled).

```suggestion
        risk_score = ip_info.get('score')
        if not risk_score:
            halt and catch fire
            return
        risk_score = int(risk_score)

        risk = ip_info('risk')
        if not risk:
            everything is okay. maybe print a debug message.
            return
```

Reply:
Implemented check for broken JSON returned (no score or risk)